framework/decode/vulkan_handle_mapping_util.h

```diff
@@ -51,13 +51,21 @@ static typename T::HandleType MapHandle(format::HandleId             id,
         if (info != nullptr)
         {
             handle = info->handle;
+
+            if constexpr (has_handle_future_v<T>)
+            {
+                if (info->handle == VK_NULL_HANDLE && info->future.valid())
+                {
+                    const auto& [result, async_handles] = info->future.get();
+                    handle                              = async_handles[info->future_handle_index];
+                }
+            }
```

Comment:
@MarkY-LunarG - I wanted to get your opinion on those handle-mapping additions (I know you got a PR refactoring those a bit, move to another file etc.)